packages/kbn-elastic-agent-utils/src/agent_guards.ts

```diff
@@ -7,32 +7,50 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { JAVA_AGENT_NAMES, OPEN_TELEMETRY_AGENT_NAMES, RUM_AGENT_NAMES } from './agent_names';
+import {
+  ANDROID_AGENT_NAMES,
+  IOS_AGENT_NAMES,
+  JAVA_AGENT_NAMES,
+  OPEN_TELEMETRY_AGENT_NAMES,
+  RUM_AGENT_NAMES,
+} from './agent_names';
 
 import type {
+  AndroidAgentName,
+  IOSAgentName,
   JavaAgentName,
   OpenTelemetryAgentName,
   RumAgentName,
   ServerlessType,
 } from './agent_names';
 
+export function hasOpenTelemetryPrefix(agentName?: string, language: string = '') {
```

Comment:
Maybe we could create unit tests for this function. WDYT?

Reply:
Agree, will add some testing for this

Reply:
Added testing